### ISSUE TITLE
YoLoV5 for incremental learning

### ISSUE THREAD

**Issue by Whisht** [ultralytics/yolov5#10184]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I am developing a incremental learning pipeline based on YOLOv5s with a multiple-head configuration where each task have a head responding for the classification and regression. In simple case, supposing that base task has `1` object, and another task, the novel one, has `1` different object. Now I want to add a new detect head to original `Detect()` module while freezing all other modules in original task, including backbone, head, and detect module for task 1.

```python
# old one
self.m = nn.ModuleList(nn.Conv2d(x, self.no * self.na, 1) for x in ch)  # output conv
# incremental one
self.m_inc = nn.ModuleList(nn.Conv2d(x, self.no * self.na, 1) for x in ch # output conv
```

Now, what I am confused is that what is the meaning of each dimension in the output of [`self.m`](https://github.com/ultralytics/yolov5/blob/master/models/yolo.py#L59). For `1` class object detection, the output shold have `3*6` dimentsions for the second axis. Let's note it as `o = self.m[0](x[0])`.  Then `o`  was splited and permuted. So I cannot have a clear understanding to the physical meaning of each dimension. Thus, I do not have a proper way to cat the output of `self.m_inc` to `o`.

A coarse version of my choice is directly stack to the output of `self.m`:
```python
x_base = self.m[i](x[i])  # conv
x_increment = self.m_increment[i](x[i])
x[i] = torch.cat((x_increment, x_base), dim=1)
```.
But the result after nms is a disaster with huge numbers of redundant bboxes.
![Uploading image.png…]()

Do you have any suggestion to do a more interpreterable `concat`? Thanks.

### Additional

_No response_

**Comment by glenn-jocher** [comment]
@Whisht hi there! It's great to hear about your incremental learning pipeline based on YOLOv5! 😊 The `self.m` output in YOLOv5 has dimensions representing grids, anchors, and objectness. When stacking the output of `self.m_inc` to `o`, it's important to understand the dimensions and properly concatenate the tensors to avoid issues with redundant bounding boxes. Understanding the output dimensions is key. I'd recommend checking out the YOLOv5 documentation for more details on interpreting the output dimensions and improving the concatenation process. Perhaps consulting the YOLO community forums can also provide valuable insights. Keep up the great work!
